scc/src/opts.rs

```diff
@@ -0,0 +1,147 @@
+use bpaf::*;
+use std::ffi::OsString;
+use std::path::PathBuf;
+use std::collections::VecDeque;
+
+#[cfg(test)]
+use std::{println as info};
+
+pub fn fix_args(args: Vec<String>) -> Vec<String> {
+    let mut fixed_args: Vec<String> = vec![];
+    let mut broken = false;
+
+    #[cfg(test)]
+    info!("Raw args: {:#?}", args);
+
+    // when cyberpunk's args are processed, the \" messes up the grouping, so we need to fix the args that have quotes & spaces in them
+    for arg in args {
+        let contains_quote = arg.contains('"');
+        if !contains_quote && !broken {
+            fixed_args.push(arg.clone());
+        } else if !broken {
+            let slashful = arg.replace('"', "\\\"");
+            for part in slashful.split("\" ") {
+                fixed_args.push(part.to_string());
+            }
+        } else {
+            // fixes args that come after the first
+            let broken_arg = if contains_quote {
+                arg.replace('"', "\\")
+            } else {
+                arg
+            };
+            if broken_arg.contains(' ') {
+                let mut parts = broken_arg.split(' ').collect::<VecDeque<&str>>();
+                let last = fixed_args.last().unwrap();
+                *fixed_args.last_mut().unwrap() = format!("{} {}", last, &parts.pop_front().unwrap());
+                for part in parts {
+                    fixed_args.push(part.to_string());
+                }
+            } else {
+                let last = fixed_args.last().unwrap();
+                *fixed_args.last_mut().unwrap() = format!("{} {}", last, broken_arg);
+            }
+        }
+        if contains_quote {
+            broken = !broken;
+        }
+    }
+
+    #[cfg(test)]
+    info!("Fixed args: {:#?}", fixed_args);
+
+    fixed_args
+}
+
+#[derive(Clone, Debug)]
+pub struct Opts {
+    pub script_paths: Vec<PathBuf>,
+    pub cache_file: PathBuf,
+    pub cache_dir: Option<PathBuf>,
+    pub optimize: bool,
+    pub threads: Option<u8>,
+    pub no_warnings: bool,
+    pub no_testonly: bool,
+    pub no_breakpoint: bool,
+    pub profile_off: bool
+}
+
+fn toggle_options(name: &'static str, help: &'static str) -> impl Parser<bool> {
+    any::<String>(name)
+        .help(help)
+        .guard(|s| s.starts_with('-'), "doesn't start with -")
+        .parse(move |s| {
+            let (state, cur_name) = if let Some(rest) = s.strip_prefix('-') {
+                (true, rest)
+            } else {
+                return Err(format!("{} is not a toggle option", s));
+            };
```

Comment:
```suggestion
            let Some(cur_name) = s.strip_prefix('-') else {
                return Err(format!("{} is not a toggle option", s));
            };
```
`state` is always true